presto-geospatial/src/main/java/com/facebook/presto/geospatial/GeoFunctions.java

```diff
@@ -0,0 +1,506 @@
+    @ScalarFunction("ST_IsClosed")
+    @SqlType(StandardTypes.BOOLEAN)
+    public static Boolean stIsClosed(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = getGeometry(deserialize(input), "ST_IsClosed", LINE_STRING, MULTI_LINE_STRING);
+        MultiPath lines = (MultiPath) geometry.getEsriGeometry();
+        int pathCount = lines.getPathCount();
+        boolean result = true;
+        for (int i = 0; result && i < pathCount; i++) {
+            Point start = lines.getPoint(lines.getPathStart(i));
+            Point end = lines.getPoint(lines.getPathEnd(i) - 1);
+            result = result && end.equals(start);
+        }
+        return result;
+    }
+
+    @SqlNullable
+    @Description("Returns true if this Geometry is an empty geometrycollection, polygon, point etc")
+    @ScalarFunction("ST_IsEmpty")
+    @SqlType(StandardTypes.BOOLEAN)
+    public static Boolean stIsEmpty(@SqlType(NAME) Slice input)
+    {
+        return deserialize(input).isEmpty();
+    }
+
+    @Description("Returns the 2d length of the geometry if it is a linestring or multilinestring")
+    @ScalarFunction("ST_Length")
+    @SqlType(StandardTypes.DOUBLE)
+    public static double stLength(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = getGeometry(deserialize(input), "ST_Length", LINE_STRING, MULTI_LINE_STRING);
+        return geometry.getEsriGeometry().calculateLength2D();
+    }
+
+    @Description("Returns X maxima of a bounding box 2d or 3d or a geometry")
+    @ScalarFunction("ST_XMax")
+    @SqlType(StandardTypes.DOUBLE)
+    public static double stXMax(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = deserialize(input);
+        Envelope envelope = getEnvelope(geometry);
+        return envelope.getXMax();
+    }
+
+    @Description("Returns Y maxima of a bounding box 2d or 3d or a geometry")
+    @ScalarFunction("ST_YMax")
+    @SqlType(StandardTypes.DOUBLE)
+    public static double stYMax(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = deserialize(input);
+        Envelope envelope = getEnvelope(geometry);
+        return envelope.getYMax();
+    }
+
+    @Description("Returns X minima of a bounding box 2d or 3d or a geometry")
+    @ScalarFunction("ST_XMin")
+    @SqlType(StandardTypes.DOUBLE)
+    public static double stXMin(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = deserialize(input);
+        Envelope envelope = getEnvelope(geometry);
+        return envelope.getXMin();
+    }
+
+    @Description("Returns Y minima of a bounding box 2d or 3d or a geometry")
+    @ScalarFunction("ST_YMin")
+    @SqlType(StandardTypes.DOUBLE)
+    public static double stYMin(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = deserialize(input);
+        Envelope envelope = getEnvelope(geometry);
+        return envelope.getYMin();
+    }
+
+    @Description("Return the number of interior rings of the first polygon in the geometry")
```

Comment:
nit: perhaps, borrow the documentation from the standard and clarify that this function only applies to polygons

"Returns the cardinality of the collection of interior rings of a polygon"